transports/quic/tests/smoke.rs

```diff
@@ -236,6 +237,57 @@ fn new_tcp_quic_transport() -> (PeerId, Boxed<(PeerId, StreamMuxerBox)>) {
     (peer_id, transport)
 }
 
+#[cfg(feature = "async-std")]
+#[test]
+fn test_address_translation_async_std() {
+    test_address_translation::<quic::async_std::Provider>()
+}
+
+#[cfg(feature = "tokio")]
+#[test]
+fn test_address_translation_tokio() {
+    test_address_translation::<quic::tokio::Provider>()
+}
+
+fn test_address_translation<T>()
+where
+    T: Provider,
+{
+    let (_peer_id, transport) = create_default_transport::<T>();
+
+    let port = 42;
+    let quic_listen_addr = Multiaddr::empty()
+        .with(Protocol::Ip4(Ipv4Addr::new(127, 0, 0, 1)))
+        .with(Protocol::Udp(port))
+        .with(Protocol::QuicV1);
```

Comment:
Perhaps a helper function that takes in an `Ipv4Addr` and a port would make this test a bit more concise.